include/osquery/killswitch.h

```diff
@@ -0,0 +1,55 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <boost/core/noncopyable.hpp>
+#include <osquery/expected.h>
+#include <osquery/status.h>
+#include <string>
+
+namespace osquery {
+
+class Killswitch : private boost::noncopyable {
+ private:
+  Killswitch();
+
+ public:
+  ~Killswitch();
+
+  /*
+  // Author: @guliashvili
+  // Creation Time: 11 / 07 / 2018
+  bool isMyTutorialFeatureEnabled();
+  */
+
+  static Killswitch& get() {
+    static Killswitch killswitch;
+    return killswitch;
+  }
+
+  Status refresh();
+
+ private:
+  bool isNewCodeEnabled(const std::string& key);
+
+  enum class IsEnabledError {
+    CallFailed = 1,
+    IncorrectResponseFormat = 2,
+    IncorrectValue = 3
+  };
+  Expected<bool, Killswitch::IsEnabledError> isEnabled(const std::string& key);
```

Comment:
How to use a `Killswitch` if this method is private? Probably I have wrong assumption about it. This a good reason to have some short description how to use it with small example. What do you think?

Reply:
pull request comment was updated after your first review. As it is in the comments of killswitch.h and killswitch.cpp(and written in the comment over there), for each killswitch new method should be created calling isEnabled. So we want to leave it as private.